vm/internal/setup/bugsnag.go

```diff
@@ -44,3 +47,32 @@ func ConfigureBugsnagHandler(server *http.Server) {
 	server.Handler = bugsnag.Handler(nil)
 	log.Info("Bugsnag handler configured successfully.")
 }
+
+func ConfigureBugsnagHTTPErrorHandler(err error, c echo.Context) {
+	if os.Getenv("BUGSNAG_API_KEY") != "" {
+		if he, ok := err.(*echo.HTTPError); ok {
+			if he.Code == http.StatusInternalServerError {
+				// log to container logs
+				log.Error(err)
+				// log to Bugsnag
+				_ = bugsnag.Notify(err, c.Request().Context())
+			}
+		} else {
+			// log to container logs
+			log.Error(err)
+			// log to Bugsnag
+			_ = bugsnag.Notify(err, c.Request().Context())
+		}
+	}
```

Comment:
nit: with early returns, I found the code more readable
- there are 2 conditional paths clearly identified 
- no need to look for the if when reading the code in the else block
- the happy path is aligned on the left, not surrounded by brackets on the 3rd level of indentation

I found this article interesting https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88

```suggestion
	if os.Getenv("BUGSNAG_API_KEY") == "" {
		return
	}
	
	he, ok := err.(*echo.HTTPError); 
	if ok && he.Code != http.StatusInternalServerError {
		return
	}
	
	// log to container logs
	log.Error(err)
	// log to Bugsnag
	_ = bugsnag.Notify(err, c.Request().Context())
```

Reply:
It's definitely more readable, thanks!